mod/CheckHintData.cs

```diff
@@ -0,0 +1,124 @@
+            get
+            {
+                if (normalTextMat == null)
+                    normalTextMat = Locator.GetAstroObject(AstroObject.Name.TimberHearth).transform
+                        .Find("Sector_TH/Sector_Village/Sector_Observatory/Interactables_Observatory/NomaiEyeExhibit/NomaiEyePivot/Arc_TH_Museum_EyeSymbol/Arc 1")
+                        .GetComponent<Renderer>().material;
+                return normalTextMat;
+            }
+        }
+
+        private static Material childTextMat;
+        private static Material normalTextMat;
+
+        private Renderer rend;
+
+        public Color NomaiWallColor()
+        {
+            if (HasBeenFound) return FoundColor;
+            switch (Importance)
+            {
+                case CheckImportance.Junk:
+                    return JunkColor;
+                case CheckImportance.Useful:
+                    return UsefulColor;
+                case CheckImportance.Progression:
+                    return ProgressionColor;
+                default:
+                    {
+                        int rnd = Random.Range(0, 3);
+                        switch (rnd)
+                        {
+                            case 0:
+                                return JunkColor;
+                            case 1:
+                                return UsefulColor;
+                            default:
+                                return ProgressionColor;
+                        }    
+                    }
+            }
+        }
+
+        public void SetImportance(CheckImportance importance)
+        {
+            if ((int)Importance < (int)importance)
+            {
+                Importance = importance;
+            }
+        }
+
+        public void DetermineImportance(Location loc)
+        {
+            rend = GetComponent<Renderer>();
+
+            Locations.Add(loc);
+            if (APRandomizer.APSession.Locations.AllLocationsChecked.Contains(LocationNames.locationToArchipelagoId[loc])) HasBeenFound = true;
+
+            if (Importance != CheckImportance.Trap)
+            {
+                if (rend.material.name.Contains("TextChild")) IsChildText = true;
+            }
+
+            switch (Scouter.ScoutedLocations[loc].Flags)
+            {
+                case ItemFlags.None:
+                    SetImportance(CheckImportance.Junk);
+                    break;
+                case ItemFlags.NeverExclude:
+                    SetImportance(CheckImportance.Useful);
+                    break;
+                case ItemFlags.Advancement:
+                    SetImportance(CheckImportance.Progression);
+                    break;
+                case ItemFlags.Trap:
+                    SetImportance(CheckImportance.Trap);
+                    rend.material = IsChildText ? NormalTextMat : ChildTextMat;
+                    break;
+            }
+        }
+    }
+    public enum CheckImportance
+    {
+        Trap = 0,
+        Junk = 1,
```

Comment:
Can we use "filler" here instead? Since AP's Python APIs call it `ItemClassification.filler` that's the term the .apworld code uses, and I've already used the term "junk" in the .apworld to mean "filler _or trap_".

Reply:
Certainly, done